algorithms/kernel/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i

```diff
@@ -144,145 +114,119 @@ Status KNNClassificationPredictKernelUCAPI<algorithmFpType>::
         for (size_t dblock = 0; dblock < nDataBlocks; dblock++)
         {
             const size_t dfirst = dblock * dataBlockSize;
-            size_t dlast = dfirst + dataBlockSize;
+            size_t dlast        = dfirst + dataBlockSize;
 
             if (dlast > nDataRows)
+            {
                 dlast = nDataRows;
+            }
             const size_t curDataBlockSize = dlast - dfirst;
 
+            BlockDescriptor<int> labelRows;
+            DAAL_CHECK_STATUS_VAR(labels->getBlockOfRows(dfirst, dataBlockSize, readOnly, labelRows));
+            auto dataLabels = labelRows.getBuffer();
+            BlockDescriptor<algorithmFpType> dataRows;
+            DAAL_CHECK_STATUS_VAR(points->getBlockOfRows(dfirst, curDataBlockSize, readOnly, dataRows));
+            auto curData = dataRows.getBuffer();
+
+            // Collect sums of squares from train data
+            auto sumResult = math::SumReducer::sum(math::Layout::RowMajor, curData, curDataBlockSize, nFeatures, &st);
+            DAAL_CHECK_STATUS_VAR(st);
+            // Initialize GEMM distances
+            initDistances(context, init_distances, sumResult.sumOfSquares, distances, curDataBlockSize, curProbeBlockSize, &st);
+            DAAL_CHECK_STATUS_VAR(st);
             // Let's calculate distances using GEMM
-            {
-                DAAL_ITTNOTIFY_SCOPED_TASK(compute.copy_data);
-                context.copy(data_temp, 0, data_all, dfirst * nFeatures, nFeatures * curDataBlockSize, &st);
-            }
-            auto data = data_temp.template get<algorithmFpType>();
-            auto sumResult = math::SumReducer::sum(math::Layout::RowMajor,
-                                                   data, curDataBlockSize, nFeatures, &st); DAAL_CHECK_STATUS_VAR(st);
-            initDistances(context, init_distances, sumResult.sumOfSquares, distances, curDataBlockSize, curProbeBlockSize, &st); DAAL_CHECK_STATUS_VAR(st);
-            computeDistances(context, data, probes, distances, curDataBlockSize, curProbeBlockSize, nFeatures, &st); DAAL_CHECK_STATUS_VAR(st);
-            {
-                DAAL_ITTNOTIFY_SCOPED_TASK(compute.copy_categories);
-                initCategories(context, init_categories, labels_all, categories, curProbeBlockSize, curDataBlockSize, dfirst, &st); DAAL_CHECK_STATUS_VAR(st);
-            }
-            QuickSelectIndexed::select( distances, categories, rndSeq, dataBlockSize, nK, curProbeBlockSize,
-                                        curDataBlockSize, curDataBlockSize, selectResult, &st);  DAAL_CHECK_STATUS_VAR(st);
-            copyPartialSelections(context, gather_selection, selectResult.values, selectResult.indices, partialDistances,
-                                  partialCategories, curProbeBlockSize, nK, partCount, nParts, dblock != 0 && partCount == 0, &st);  DAAL_CHECK_STATUS_VAR(st);
+            computeDistances(context, curData, curProbes, distances, curDataBlockSize, curProbeBlockSize, nFeatures, &st);
```

Comment:
From this point of view, it is unclear what is the difference between `initDistances` and `computeDistances`. Can we provide more clear names here?